install_config/router/customized_haproxy_router.adoc

```diff
@@ -13,48 +13,673 @@ toc::[]
+|===
+|Field | Type | Description
+|ID | string|
+|Contents| string| The certificate
+|PrivateKey|string| The private key
+|===
+
+.Router TLSTerminationType
+[cols="3*", options="header"]
+|===
+|Field | Type | Description
+|`*TLSTerminationType*` | string | TLSTerminationType dictates where the
+secure communication will stop
+|`*InsecureEdgeTerminationPolicyType*` | string | insecureEdgeTerminationPolicy
+indicates the desired behavior for insecure connections to a route. While
+each router may make its own decisions on which ports to expose, this is
+normally port 80.
+|===
+
+TLSTerminationType and InsecureEdgeTerminationPolicyType dictates where
+the secure communication will stop.
+
+
+.Router TLSTerminationType Values
+[cols="2*", options="header"]
+|===
+|Value | Meaning
+|`*edge*` | TLSTerminationEdge terminate encryption at the edge router.
+|`*passthrough*` | TLSTerminationPassthrough terminate encryption at
+the destination, the destination is responsible for decrypting traffic
+|`*reencrypt*` | TLSTerminationReencrypt terminate encryption at the edge
+router and re-encrypt it with a new certificate supplied by the destination
+|===
+
+.Router InsecureEdgeTerminationPolicyType Values
+[cols="2*", options="header"]
+|===
+|Type | Meaning
+|`*Allow*` | traffic is sent to the server on the insecure port (default)
+|`*Disable*` | no traffic is allowed on the insecure port.
+|`*Redirect*` | clients are redirected to the secure port.
+|===
+None ("") is the same as `*Disable*`.
+
+[[using-annotations]]
+=== Annotations
+
+Each route can have annotations attached. The Annotations are just a name
+and a value.
+
+The name can be pretty much anything that doesn't conflict with existing
+Annotations.  The value is any string. The string can have multiple tokens
+separated by a space e.g., "aa bb cc".  The template uses {{index}} to
+extract the value of an annotation.
+
+.xref:../../architecture/core_concepts/routes.adoc#route-specific-annotations[Route-specific Annotations]
+[cols="3*", options="header"]
+|===
+|Annotation | Test | Default
+|haproxy.router.openshift.io/balance | |
+|haproxy.router.openshift.io/timeout | [1-9][0-9]*(us ms s m h d)? | none set
+|haproxy.router.openshift.io/rate-limit-connections |"true" "TRUE" | ""
+|haproxy.router.openshift.io/rate-limit-connections.concurrent-tcp | | ""
+|haproxy.router.openshift.io/rate-limit-connections.rate-tcp | | ""
+|haproxy.router.openshift.io/rate-limit-connections.rate-http | | ""
+|router.openshift.io/haproxy.health.check.interval |[1-9][0-9]*(us ms s m h d)? |5000ms
+|haproxy.router.openshift.io/disable_cookies |true TRUE |
+|===
+The above annotations are being used by the HAProxy router. haproxy.router.openshift.io/rate-limit-connections
+must be "true" for the other rate-limit-connections to work. The annotation values must be quoted e.g.,
+haproxy.router.openshift.io/rate-limit-connections.rate-tcp: "66".
+
+[[using-env-var]]
+=== Environment Variables
```

Comment:
@JacobTanenbaum @knobunc there are now three discussions about annotations and each takes a different slant. architecture/core_concepts/routes.html has "Route-specific Annotations", nstall_config/router/default_haproxy_router.htm has "Protecting Against DDoS Attacks", and install_config/router/customized_haproxy_router.html has "Annotations". I will delete the table in the customized_haproxy_router and point back to the ddos discussion.

The environment variables are in two places a complete list in "architecture/core_concepts/routes.html" with description and defaults. The "install_config/router/customized_haproxy_router.html" discussion lists the ones that are currently used in the tamplate along with how they are used. I added these in the spirit of knowing what is there and helping to decide whether to modify the template of provide a new one. We could add more columns to the table in "architecture/core_concepts/routes.html" but that table is getting wide as is and this would make it wider. For now I will leave the env vars as is, pending further discusion.